Cognoma.py

```diff
@@ -0,0 +1,656 @@
+                                               then encoded and decoded as
+                                               left.
+            left_generatedImgs (array of floats): Random encoded points
+                                                    decoded as left.
+            rightToLeftImgs (array of floats): Right input encoded and decoded
+                                               as left.
+            params (model_parameters): Parameters of the model.
+            n (int): Defaults to 100
+
+        Returns: None
+        """
+
+        #######################################################################
+        # For calculating the t-test between the MSE of real and generated data
+
+        # Find all TP53 wildtype examples and their gene expressions
+        SNPpresent = np.array(rightDomain[0, :])
+        SNPpresentExp = np.array(leftDomain[0, :])
+
+        for x in range(rightDomain.shape[0]):
+            if rightDomain[x, self.mutationID] == 0:
+                SNPpresent = np.vstack((SNPpresent, rightDomain[x, :]))
+                SNPpresentExp = np.vstack(
+                    (SNPpresentExp, leftDomain[x, :]))
+
+        # Find all TP53 mutated examples and their gene expressions
+        SNPabsent = np.array(rightDomain[0, :])
+        SNPabsentExp = np.array(leftDomain[0, :])
+
+        for x in range(rightDomain.shape[0]):
+            if rightDomain[x, self.mutationID] == 1:
+                SNPabsent = np.vstack((SNPabsent, rightDomain[x, :]))
+                SNPabsentExp = np.vstack((SNPabsentExp, leftDomain[x, :]))
+
+        # Pick a random grouping of these examples
+        randIndexes = np.random.randint(
+            0, min(SNPpresent.shape[0], SNPabsent.shape[0]) - 1,
+            (n,))
+        SNPpresent = SNPpresent[1:]
+        SNPpresentExp = SNPpresentExp[1:]
+        SNPpresent = SNPpresent[randIndexes]
+        SNPpresentExp = SNPpresentExp[randIndexes]
+
+        SNPabsent = SNPabsent[1:]
+        SNPabsentExp = SNPabsentExp[1:]
+        SNPabsent = SNPabsent[randIndexes]
+        SNPabsentExp = SNPabsentExp[randIndexes]
+
+        # Predict based on the mutations
+        (SNPpresentToExp, _) = modelHandle.rightToLeftModel.predict(
+                                                              SNPpresent)
+
+        # Find the Mean Squared Error
+        SNPpresentToExpMSE = np.square(np.subtract(
+            SNPpresentExp, SNPpresentToExp)).mean(axis=0)
+
+        # Create arrays for induced samples
+        InducedMutation = copy.copy(SNPpresent)
+        InducedMutationExp = copy.copy(SNPpresentExp)
+
+        # Induce a TP53 mutation
+        InducedMutation[:, self.mutationID] = 1
+
+        # Predict based on this mutation
+        (InducedMutationToExp, _) = modelHandle.rightToLeftModel.predict(
+                                                             InducedMutation)
+
+        # Find the Mean Squared Error
+        InducedMutationToExpMSE = np.square(np.subtract(
```

Comment:
can we use the sklearn implementation? http://scikit-learn.org/stable/modules/generated/sklearn.metrics.mean_squared_error.html